compiler/src/dotty/tools/dotc/transform/Splicing.scala

```diff
@@ -242,9 +242,14 @@ class Splicing extends MacroTransform:
                 else args.mapConserve(arg => transformLevel0QuoteContent(arg)(using quoteContext))
               }
               cpy.Apply(tree)(cpy.Select(sel)(cpy.Apply(app)(fn, newArgs), nme.apply), quotesArgs)
-        case Apply(TypeApply(_, List(tpt)), List(quotes))
+        case Apply(TypeApply(typeof, List(tpt)), List(quotes))
         if tree.symbol == defn.QuotedTypeModule_of && containsCapturedType(tpt.tpe) =>
-          ref(capturedType(tpt))(using ctx.withSource(tree.source)).withSpan(tree.span)
+          tpt match
+            case block: Block =>
+              val newBlock = capturedBlockPartTypes(block)
+              Apply(TypeApply(typeof, List(newBlock)), List(quotes)).withSpan(tree.span)
```

Comment:
To be safe, we should also create the `Apply`/`TypeApply` using `ctx.withSource(tree.source)`. Probably worth wrapping this case with `inContext(ctx.withSource(tree.source)) { ... }`.